docs/setup-guide/multi-node-dev-cluster.md

```diff
@@ -1,52 +1,73 @@
 
-This doc, as at the moment written, does not mandate which cni plugin to be used. It is up to user, though we recommend to use calico, or our own alktron.
+This doc, as at the moment written, does not mandate which cni plugin to be used; it is up to the user. We have verified that Flannel works well with multi-tenancy Arktos cluster; the instructions laid out here is based on our experience with Flannel at GCP env.
 
-1. start the master node 
- the master node, run follwoing commands to start cluster, which has the master only.
+1. bootstrap the cluster by starting the master node (no CNI plgin at first)
 ```bash
+export ARKTOS_NO_CNI_PREINSTALLED=y
 ./hack/arktos-up.sh
 ```
 
-2. copy needed files to worker node
+2. install CNI plugin
+```bash
+./cluster/kubectl.sh apply -f https://github.com/coreos/flannel/raw/master/Documentation/kube-flannel.yml
+```
 
-following files are required to copy over from master to worker node:
-_output/local/bin/linux/amd64/hyperkube
-/var/run/kubernetes/kubelet.kubeconfig
-/var/run/kubernetes/kube-proxy.kubeconfig
-/var/run/kubernetes/client-ca.crt
-/tmp/kube-proxy.yaml
+3. on the GCP instance to be added as worker node, ensure follwoing worker secret files copied from the master node:
+```bash
+mkdir -p /tmp/arktos
+gcloud compute scp <master-node-instance>:/var/run/kubernetes/kubelet.kubeconfig /tmp/arktos/
+gcloud compute scp <master-node-instance>:/var/run/kubernetes/client-ca.crt /tmp/arktos/
+```
 
+At the moment this doc is written, kube-proxy does not support multi-tenancy yet, won't be deployed here. After kube-proxy issue has been fixed (we will allocate resource to cope with it soon), probably we will also need to copy over kube-proxy related secret and artifacts.
 
-3. start worker node
-at worker node, run follwoing commands
+As a temporary measure to accommodate Flannel pods to access api server through the kubernetes service IP (the notorious 10.0.0.1) at the absense of kube-proxy, we need to place a bit trick by running below command at the worker node, assuming the kupe-apiserver is listening at https://10.138.0.19:6443:
 ```bash
-export KUBELET_IP=<worker-ip>
-export KUBELET_KUBECONFIG=<path-to-kubelet.kubeconfig-copied-from-master>
-./arktos-worker-up.sh
+sudo iptables -t nat -A OUTPUT -p tcp -d 10.0.0.1 --dport 443 -j DNAT --to-destination 10.138.0.19:6443
 ```
 
-If kube-proxy is desired to run on worker, first make sure kube-proxy.yaml points to proper kube-proxy.kubeconfig, then run (note: for alktron as cni plugin, you don't need to run kube-proxy at all)
+Please be advised that this is a temporary quirk only; after we have the full service support by proper kube-proxy or other means, we don't need it any more.
+
+4. start the worker node and register into cluster
+
+at worker node, run following commands:
```

Comment:
I think both the work node(s) and master node need to download the arktos repo. If that is the case, it is better to tell the user to do it first.

Reply:
will make this assumption explicit